### PR TITLE
HHH-15634 Lazy basic property does not get updated on change

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15634

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java [hunk 1]
```diff
@@ -3852,6 +3852,24 @@ public void update(
...
 		}
+		else if ( dirtyFields != null && hasUninitializedLazyProperties( object ) && hasLazyDirtyFields( dirtyFields ) ) {
+			// We need to generate the UPDATE SQL when there are dirty lazy fields
+			propsToUpdate = getPropertiesToUpdate( dirtyFields, hasDirtyCollection );
+			// don't need to check laziness (dirty checking algorithm handles that)
+			final boolean[] propertyLaziness = getPropertyLaziness();
+			// we add also all the non lazy properties because dynamic update is false
+			for ( int i = 0; i < propertyLaziness.length; i++ ) {
+				if ( propertyLaziness[i] == false ) {
+					propsToUpdate[i] = true;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java:3863]: We should probably have a quick discussion about this next week with @sebersole to hear his thoughts on this topic.

**Review by beikov — COMMENTED** [review]
Seems that you accidently also fixed https://hibernate.atlassian.net/browse/HHH-10747 since a failure expected test `org.hibernate.orm.test.bytecode.enhancement.lazy.LazyLoadingByEnhancerSetterTest` is now running through 😄

**Review by yrodiere — APPROVED** [review]
I ran the reproducer in Quarkus (https://github.com/quarkusio/quarkus/pull/28881) against Hibernate ORM 5.6 compiled with a backport of this patch (#5465), and I confirm it fixes all issues.

Thanks a lot @dreab8!

**Comment by dreab8** [comment]
@beikov gave me a huge help :) by the way we have to check that the call of the `InlineDirtyCheckerEqualsHelper` does not create any issue

**Comment by yrodiere** [comment]
Then thanks to @beikov too :)

**Pull request merged**